pkg/generator_test.go

```diff
@@ -337,6 +337,22 @@ func (s *GeneratorSuite) TestGeneratorChecksInterfacesForNilable() {
 	s.checkGeneration("requester_iface.go", "RequesterIface", false, "", "")
 }
 
+func (s *GeneratorSuite) TestGeneratorChecksInterfacesWithGenericInterfaceTypedParamForNilable() {
+	expectedBytes, err := os.ReadFile(getMocksPath("RequesterIfaceWithInterfaceTypedParam.go"))
+	s.Require().NoError(err)
+
+	expected := string(expectedBytes)
+	expected = expected[strings.Index(expected, "// RequesterIfaceWithInterfaceTypedParam is"):]
+
+	cfg := GeneratorConfig{
+		StructName:     "RequesterIfaceWithInterfaceTypedParam",
+		WithExpecter:   true,
+		UnrollVariadic: false,
+	}
+
+	s.checkGenerationWithConfig("requester_iface.go", "RequesterIfaceWithInterfaceTypedParam", cfg, expected)
+}
+
```

Comment:
Instead of this, I'd prefer to run an actual test on the generated mock itself. Like, run a test that checks that we can supply a nil value as the return and that we get the nil interface back. This test you have here is just doing a simple text comparison, which is not useful.

This might be a bit awkward if the generated mock is placed in the separate `mocks/` directory. Instead, generate it in the same path as the fixture itself to prevent cyclic imports.

Reply:
Hello 👋  I've applied the changes in a separate commit so you're able to see what I've changed. Looking forward to hearing from you to know if it fulfills the purpose or should I apply any other small tweak